src/services/taskService.js

```diff
@@ -0,0 +1,51 @@
+const KEYS = {
+    tasks: 'tasks',
+    taskId: 'taskId',
+}
+
+export function insertTask(data) {
+    let tasks = getAllTasks();
+    data['id'] = generateTaskId();
+    tasks.push(data);
+    localStorage.setItem(KEYS.tasks, JSON.stringify(tasks));
+}
+
+export function updateTask(data) {
+    let tasks = getAllTasks();
+    let recordIndex = tasks.findIndex(x => x.id == data.id);
+    tasks[recordIndex] = { ...data };
+    localStorage.setItem(KEYS.tasks, JSON.stringify(tasks));
+}
+
+export function generateTaskId() {
+    if (localStorage.getItem(KEYS.taskId) == 0) {
```

Comment:
as a practice, avoid using incremental, index based id's.  
  
You can create a unique ID using JS' built in library:  
https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID.  
  
An sortable, time-indexed id may also be useful sometimes, you can read about ULID.  
https://blog.bitsrc.io/ulid-vs-uuid-sortable-random-id-generators-for-javascript-183400ef862c